celery/app/trace.py

```diff
@@ -356,6 +360,7 @@ def trace_task(uuid, args, kwargs, request=None):
             task_request = Context(request or {}, args=args,
                                    called_directly=False, kwargs=kwargs)
             root_id = task_request.root_id or uuid
+            the_name = request['shadow'] if 'shadow' in request else name
```

Comment:
I might be wrong here but isn't this the equivalent of:

```
trace_name = task_request.shadow if hasattr('shadow', task_request) else name
```

Reply:
Got it. Yes, if we use `task_request`, we can combine the 2 parts. :-)